app/src/common/shared/org/mozilla/vrbrowser/ui/widgets/WindowWidget.java

```diff
@@ -1095,21 +1100,35 @@ public void draw(Canvas aCanvas) {
         }
     }
 
-    public void setSession(@NonNull Session aSession) {
-        setSession(aSession, SESSION_RELEASE_DISPLAY);
+    public void setSession(@NonNull Session aSession, @SetSessionActiveState int aSessionState) {
+        setSession(aSession, SESSION_RELEASE_DISPLAY, aSessionState);
     }
 
-    public void setSession(@NonNull Session aSession, @OldSessionDisplayAction int aDisplayAction) {
+    public void setSession(@NonNull Session aSession, @OldSessionDisplayAction int aDisplayAction, @SetSessionActiveState int aSessionState) {
         if (mSession != aSession) {
             Session oldSession = mSession;
             if (oldSession != null) {
                 cleanListeners(oldSession);
+                switch (aSessionState) {
+                    case ACTIVATE:
+                        mSession.setActive(true);
+                        break;
+                    case DEACTIVATE:
+                        mSession.setActive(false);
+                        break;
+                    case DO_NOT_CHANGE:
+                        break;
+                }
                 if (aDisplayAction == SESSION_RELEASE_DISPLAY) {
                     oldSession.releaseDisplay();
                 }
             }
 
             mSession = aSession;
+
+            setupListeners(mSession);
+            SessionStore.get().setActiveSession(mSession);
```

Comment:
SessionStore active session is set here regardless if the session itself is set to DEACTIVE. Is this correct?

Reply:
The **setSession** target session is always activated, what the **aSessionState** parameter determines is what we do with the old session.